### ISSUE TITLE
json_dumps should be a part of the class to make overriding it easy

### ISSUE THREAD

**Issue by ashwch** [mixpanel/mixpanel-python#56]
Currently [`json_dumps`](https://github.com/mixpanel/mixpanel-python/blob/master/mixpanel/__init__.py#L39) is a module level method which makes it difficult to override it.

It should be a class level method so that base classes can easily override it.

For example let's say I want to handle `decimal.Decimal` fields as well then currently I will have to do something like this:

```
from decimal import Decimal
import json
import mixpanel

class MixPanelDataSerializer(mixpanel.DatetimeSerializer):
    def default(self, obj):
        if isinstance(obj, Decimal):
            return float(obj)
        return super(MixPanelDataSerializer, self).default(obj)


def json_dumps(data):
    return json.dumps(data, separators=(',', ':'), cls=MixPanelDataSerializer)

mixpanel.json_dumps = json_dumps  # assign the new function to the module object
```

What do you guys think?

**Comment by zakj** [comment]
Great idea. I'm inclined to add an optional JSON serializer argument to `Mixpanel.__init__` directly, to save you having to also subclass `Mixpanel` to override `json_dumps`. I can't promise when we'll have time to add this feature, but it's on my radar!

**Comment by ashwch** [comment]
@zakj That sounds like a good idea, I've submitted a patch for the same: https://github.com/mixpanel/mixpanel-python/pull/58

**Comment by ashwch** [comment]
@zakj Any update on this?

**Comment by zakj** [comment]
Sorry @ashwch, I haven't had time this week to look at your PR. I'll try to get a new release out next week.

**Comment by ashwch** [comment]
Hi @zakj, any update on this?

**Comment by zakj** [comment]
Only a month behind schedule! Fixed in 355f99605559cd48f43cc459b7f48db65fc40e69
